src/test/test-string-util.c

```diff
@@ -868,43 +879,187 @@ TEST(strverscmp_improved) {
+        test_strverscmp_improved_newer("123.45-67.ab", "123.45-67.89");
+        test_strverscmp_improved_newer("123.45-67.9", "123.45-67.89");
+        test_strverscmp_improved_newer("123.45-67", "123.45-67.89");
+        test_strverscmp_improved_newer("123.45-66.89", "123.45-67.89");
+        test_strverscmp_improved_newer("123.45-9.99", "123.45-67.89");
+        test_strverscmp_improved_newer("123.42-99.99", "123.45-67.89");
+        test_strverscmp_improved_newer("123-99.99", "123.45-67.89");
 
         /* '~' : pre-releases */
-        test_strverscmp_improved_one("123.45-67.89", "123~rc1-99.99");
-        test_strverscmp_improved_one("123-45.67.89", "123~rc1-99.99");
-        test_strverscmp_improved_one("123~rc2-67.89", "123~rc1-99.99");
-        test_strverscmp_improved_one("123^aa2-67.89", "123~rc1-99.99");
-        test_strverscmp_improved_one("123aa2-67.89", "123~rc1-99.99");
+        test_strverscmp_improved_newer("123~rc1-99.99", "123.45-67.89");
+        test_strverscmp_improved_newer("123~rc1-99.99", "123-45.67.89");
+        test_strverscmp_improved_newer("123~rc1-99.99", "123~rc2-67.89");
+        test_strverscmp_improved_newer("123~rc1-99.99", "123^aa2-67.89");
+        test_strverscmp_improved_newer("123~rc1-99.99", "123aa2-67.89");
 
         /* '-' : separator between version and release. */
-        test_strverscmp_improved_one("123.45-67.89", "123-99.99");
-        test_strverscmp_improved_one("123^aa2-67.89", "123-99.99");
-        test_strverscmp_improved_one("123aa2-67.89", "123-99.99");
+        test_strverscmp_improved_newer("123-99.99", "123.45-67.89");
+        test_strverscmp_improved_newer("123-99.99", "123^aa2-67.89");
+        test_strverscmp_improved_newer("123-99.99", "123aa2-67.89");
 
         /* '^' : patch releases */
-        test_strverscmp_improved_one("123.45-67.89", "123^45-67.89");
-        test_strverscmp_improved_one("123^aa2-67.89", "123^aa1-99.99");
-        test_strverscmp_improved_one("123aa2-67.89", "123^aa2-67.89");
+        test_strverscmp_improved_newer("123^45-67.89", "123.45-67.89");
+        test_strverscmp_improved_newer("123^aa1-99.99", "123^aa2-67.89");
+        test_strverscmp_improved_newer("123^aa2-67.89", "123aa2-67.89");
 
         /* '.' : point release */
-        test_strverscmp_improved_one("123aa2-67.89", "123.aa2-67.89");
-        test_strverscmp_improved_one("123.ab2-67.89", "123.aa2-67.89");
+        test_strverscmp_improved_newer("123.aa2-67.89", "123aa2-67.89");
+        test_strverscmp_improved_newer("123.aa2-67.89", "123.ab2-67.89");
 
         /* invalid characters */
         assert_se(strverscmp_improved("123_aa2-67.89", "123aa+2-67.89") == 0);
 }
 
+#define RPMVERCMP(a, b, c) \
+        test_strverscmp_improved_one(STRINGIFY(a), STRINGIFY(b), (c))
+
+TEST(strverscmp_improved_rpm) {
+        /* Tests copied from rmp's rpmio test suite.
+         * The original form is retained for easy comparisons and updates. */
```

Comment:
can you add also the link and the note about the license from the commit message? easier to see it as a comment

Reply:
Done.